mux/src/tmux_commands.rs

```diff
@@ -32,22 +33,46 @@ pub(crate) struct PaneItem {
     pane_height: u64,
     pane_left: u64,
     pane_top: u64,
+    pane_active: bool,
+}
+
+#[derive(Debug)]
+enum TmuxLayout {
+    SplitVertical(Vec<PaneItem>),
+    SplitHorizontal(Vec<PaneItem>),
+    SinglePane(PaneItem),
+}
+
+#[derive(Debug)]
+struct WindowItem {
+    session_id: TmuxSessionId,
+    window_id: TmuxWindowId,
+    window_width: u64,
+    window_height: u64,
+    window_active: bool,
+    window_name: String,
+    layout: Vec<TmuxLayout>,
+    layout_csum: String,
 }
 
 impl TmuxDomainState {
     /// check if a PaneItem received from ListAllPanes has been attached
-    fn check_pane_attached(&self, target: &PaneItem) -> bool {
-        let pane_list = self.gui_tabs.lock();
-        let local_tab = match pane_list
-            .iter()
-            .find(|&x| x.tmux_window_id == target.window_id)
-        {
-            Some(x) => x,
+    pub fn check_pane_attached(&self, window_id: TmuxWindowId, pane_id: TmuxPaneId) -> bool {
+        let gui_tabs = self.gui_tabs.lock();
+        let local_tab = gui_tabs.get(&window_id).unwrap();
```

Comment:
bare unwrap here; can we eliminate it to avoid crashing wezterm?

```suggestion
        let Some(local_tab) = gui_tabs.get(&window_id) else { return false; };
```